alibuild_helpers/utilities.py

```diff
@@ -123,7 +123,7 @@ def resolve_version(spec, defaults, branch_basename, branch_stream):
   defaults_upper = defaults != "release" and "_" + defaults.upper().replace("-", "_") or ""
   commit_hash = spec.get("commit_hash", "hash_unknown")
   tag = str(spec.get("tag", "tag_unknown"))
-  return spec["version"] % {
```

Comment:
I am not sure this will work correctly. For example the float `1.00` which might be intended as the version `"1.00"` will happily convert to `"1.0"`. If you really want to fix this, it needs to be done at the level of the parser which should always consider "version" to be followed by a string.